gwtp-core/gwtp-dispatch-common-client/src/main/java/com/gwtplatform/dispatch/client/DispatchCall.java

```diff
@@ -130,32 +129,17 @@ protected String getSecurityCookie() {
      * Override this method to perform additional work when the action execution succeeded.
      *
      * @param result the action result.
-     */
-    protected void onExecuteSuccess(R result) {
-        callback.onSuccess(result);
-    }
-
-    /**
-     * Override this method to perform additional work when the action execution succeeded.
-     *
-     * @param result the action result.
      * @param response the action {@link Response}.
      */
-    protected void onExecuteSuccess(R result, Response response) {
-        onExecuteSuccess(result);
-    }
+    protected abstract void onExecuteSuccess(R result, Response response);
 
     /**
      * Override this method to perform additional work when the action execution failed.
      *
      * @param caught the caught {@link Throwable}.
      */
-    protected void onExecuteFailure(Throwable caught) {
-        if (exceptionHandler != null && exceptionHandler.onFailure(caught) == Status.STOP) {
-            return;
-        }
-
-        callback.onFailure(caught);
+    protected boolean shouldHandleFailure(Throwable caught) {
+        return  !(exceptionHandler != null && exceptionHandler.onFailure(caught) == Status.STOP);
```

Comment:
ExceptionHandler should never be null. The default binding is DefaultExceptionHandler (it always return CONTINUE)